wordpress-mobile/WordPress-Android
# Custom user roles

Fixes #4050 #5852. This PR introduces custom user roles. Previously, we were using a static list of roles for both updating the user and inviting them. https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/523 introduced custom roles to FluxC and this PR takes care of the integration.

Here is a list of changes made:
1. We store the role as `String` in `Person` model since the `/users` endpoint will only return the name of the role
2. Remove previously used `Role` enum
3. Since the `/users` endpoint doesn't return the display name of a role, we'll try to find it from the user roles of the site and if we find a match, we'll use the display name of that role, otherwise, we'll just capitialize the string. This is an unfortunate and expensive workaround, but that's what Calypso uses too.
4. `/roles` endpoint doesn't return the invite roles and it needs to include `viewer` or `follower` role depending on the privacy of the site. This is again what Calypso does.
5. Fixes a saving state issue in `PersonDetailFragment`. IIRC we retain these fragments, that's why it didn't matter.

Known issues:
1. Jetpack `/roles` endpoint is not localized

To test:
* Invite and change the role of the user. Check the names of the roles during tests. Run it for both .com and Jetpack site.
